File: p2/c++/main.cc
#include "agrupacion-estatica.h"
#include "agrupacion-dinamica.h"
#include <iostream>

//Define la cabecera de esta función mediante programacion generica para que sea equivalente
//a declarar las dos funciones siguientes: 
//void anyadir_elementos(agrupacion_estatica<int>& ag)
//void anyadir_elementos(agrupacion_dinamica<int>& ag)
template<class T>
void anyadir_elementos(T& ag)
{
	ag.anyadir(42);
	ag.anyadir(15);
	ag.borrarUltimo();
	for (int i = 0; i < 100; i+=5) ag.anyadir(i);
}

//Define la cabecera de esta función mediante programacion generica para que sea equivalente
//a declarar las dos funciones siguientes: 
//void mostrar_elementos(const agrupacion_estatica<int>& ag)
//void mostrar_elementos(const agrupacion_dinamica<int>& ag)
template<class T>
void mostrar_elementos(const T& ag)
{
	for (int i : ag) std::cout<<i<<" ";
	std::cout<<std::endl;
}

int main(int argc,char* argv[])
{
	agrupacion_estatica<int> ag_e;
	agrupacion_dinamica<int> ag_d;

	anyadir_elementos(ag_e);
	anyadir_elementos(ag_d);

	mostrar_elementos(ag_e);
	mostrar_elementos(ag_d);	
}

File: p2/java/Main.java
public class Main
{
	//TODO: Define la cabecera de este metodo aprovechando la herencia para que sea equivalente
	//a declarar los dos metodos siguientes: 
	//public static void anyadir_elementos(AgrupacionEstatica<Integer> ag)
	//public static void anyadir_elementos(AgrupacionDinamica<Integer> ag)
	public static void anyadir_elementos(Agrupacion<Integer> ag)
	{
		ag.anyadir(42);
		ag.anyadir(15);
		ag.borrarUltimo();
		for (int i = 0; i < 100; i+=5) ag.anyadir(i);
	}

	//TODO: Define la cabecera de este metodo aprovechando la herencia para que sea equivalente
	//a declarar los dos metodos siguientes: 
	//public static void mostrar_elementos(AgrupacionEstatica<Integer> ag)
	//public static void mostrar_elementos(AgrupacionDinamica<Integer> ag)
	public static void mostrar_elementos(Agrupacion<Integer> ag)
	{
		for (Integer i : ag) 
			System.out.print(i+" ");
		System.out.println();
	}

	public static void main(String[] args)
	{
		AgrupacionEstatica<Integer> ag_e = new AgrupacionEstatica<Integer>();
		AgrupacionDinamica<Integer> ag_d = new AgrupacionDinamica<Integer>();

		anyadir_elementos(ag_e);
		anyadir_elementos(ag_d);

		mostrar_elementos(ag_e);
		mostrar_elementos(ag_d);
	}
}
